### PR TITLE
ENT-9380: Added doc for policy function filereadable

### PR DESCRIPTION
**Pull request by larsewi**
Depends on: https://github.com/cfengine/core/pull/5147

### CODE REVIEW

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/9130/)

**Comment by larsewi** [comment]
@cf-bottom are you back from vacation now? If so, can I have a build in Jenkins? :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/build-and-deploy-docs-master/69/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/69/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-69/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-69/output/_site/

**Comment by nickanderson** [comment]
Pushed a couple small changes directly to your branch @larsewi.

@cf-bottom jenkins, please

**Review by nickanderson — APPROVED** [review]

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/build-and-deploy-docs-master/71/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/71/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-71/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-71/output/_site/

File: reference/functions/filereadable.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+**Example:**
+
+[%CFEngine_include_snippet(filereadable.cf, #\+begin_src cfengine3, .*end_src)%]
+
+Output:
+
+[%CFEngine_include_snippet(filereadable.cf, #\+begin_src\s+example_output\s*, .*end_src)%]
+
+**History:** Introduced in 3.21.
```
Comment (nickanderson) [inline reference/functions/filereadable.markdown:38]:
```suggestion
**History:** Introduced in 3.22.0
```

**Review by nickanderson — COMMENTED** [review]
3.21 has come and gone, so this will have to be 3.22.0 at least.

**Comment by larsewi** [comment]
@nickanderson, I think I accidentally removed your changes by force 😬 Sorry! Can't seem to find the commit in the "conversation" tab. Not sure how to fix this.

**Comment by nickanderson** [comment]
image
image

**Comment by larsewi** [comment]
@nickanderson did you not add some functions to see also?

**Comment by larsewi** [comment]
@cf-bottom Jenkins :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/build-and-deploy-docs-master/74/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/74/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-74/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-74/output/_site/

**Comment by nickanderson** [comment]
> @nickanderson did you not add some functions to see also?

Not that I recall. But `isexecutable()`, `isdir()`, `islink()`, `isplain()` come to mind right now. Looking at those function names, makes me slightly question your name choice.

**Comment by larsewi** [comment]
> > @nickanderson did you not add some functions to see also?

> Not that I recall. But isexecutable(), isdir(), islink(), isplain() come to mind right now. Looking at those function names, makes me slightly question your name choice.

Changed name and added the "see also" functions

File: reference/functions/isreadable.markdown [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+title: isreadable
+published: true
+tags: [reference, file functions, functions, files]
+---
+
+[%CFEngine_function_prototype(path, timeout)%]
+
+**Description:** Check if file is readable.
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:10]:
```suggestion
**Description:** Check if a file is readable.
```

File: reference/functions/isreadable.markdown [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This function checks if file specified by the `path` parameter is readable by
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:12]:
```suggestion
This function checks if the file specified by the `path` parameter is readable by
```

File: reference/functions/isreadable.markdown [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+trying to read 1 Byte from it. If file is not ready for I/O, and the `timeout`
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:13]:
```suggestion
trying to read 1 Byte from it. If the file is not ready for I/O, and the `timeout`
```

File: reference/functions/isreadable.markdown [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+parameter is not set to 0, the read operation will block for at most N number of
+seconds specified by the `timeout` parameter.
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:15]:
```suggestion
parameter is not set to 0, the read operation will block for at most `timeout` number of seconds.
```

File: reference/functions/isreadable.markdown [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+The `timeout` parameter is optional and defaults to 3 seconds if it is not
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:17]:
```suggestion
The `timeout` parameter is optional and defaults to 3 seconds, if it is not
```

File: reference/functions/isreadable.markdown [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+specified. If the `timeout` parameter is set to 0, the read operation may block
+indefinetly.
+
+The function evaluates to false if the time limit expires or the read operation
+failes for any other reason (e.g. permission denied).
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:22]:
```suggestion
fails for any other reason (e.g. permission denied).
```

File: reference/functions/isreadable.markdown [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+**Notes:**
+
+When the `timeout` parameter is set to 0, the agent will try to open and read
+the file in the main thread, which can cause the agent to block indefinetly.
+
+When the `timeout` parameter is not 0, the agent will spawn a separate thread
+that will try to open and read the file. The agent will wait for at most N
+number of seconds for the spawned thread to finnish. If the thread does not
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:31]:
```suggestion
number of seconds for the spawned thread to finish. If the thread does not
```

File: reference/functions/isreadable.markdown [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+finnish in time, the agent will cancel the thread execution, and concider the
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:32]:
```suggestion
finish in time, the agent will cancel the thread execution, and consider the
```

File: reference/functions/isreadable.markdown [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+file unreadable.
+
```
Comment (olehermanse) [inline reference/functions/isreadable.markdown:34]:
```suggestion

If the file is of size 0, the function will return true, if it successfully reads 0 bytes (reaches end-of-file).

```

**Review by olehermanse — APPROVED** [review]

File: reference/functions/isreadable.markdown (hunk 9 above)
Comment (nickanderson) [reply reference/functions/isreadable.markdown:32]: got something against the fins? What if  it was sweedish in time?

**Pull request merged**